apps/server/src/modules/files-storage/uc/files-storage.uc.ts

```diff
@@ -130,33 +140,35 @@ export class FilesStorageUC {
 		await this.authorizationService.checkPermissionByReferences(userId, allowedType, parentId, context);
 	}
 
-	async download(userId: EntityId, params: DownloadFileParams) {
+	private checkFileName(entity: FileRecord, params: DownloadFileParams): void | NotFoundException {
+		if (entity.name !== params.fileName) {
+			throw new NotFoundException(ErrorType.FILE_NOT_FOUND, `${FilesStorageUC.name}:download`);
+		}
+	}
+
+	private checkScanStatus(entity: FileRecord): void | NotAcceptableException {
+		if (entity.securityCheck.status === ScanStatus.BLOCKED) {
+			throw new NotAcceptableException(ErrorType.FILE_IS_BLOCKED, `${FilesStorageUC.name}:download`);
```

Comment:
I don't think `NotAcceptableException` is the correct exception here. It has something to do with content-negotiation-headers: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/406

I would suggest `BadRequestException` or `UnprocessableEntityException`. I looked for a recommendation for the status code in case of virus detection, but I couldn't really find one...

Reply:
I only move the code and not touch the type of the exception. BadRequest and Unprocessable sounds also wrong. 

"Der 422 Unprocessable Entity HTTP Status Code bedeutet, dass der Server den Content-Type der Anfrage Entität verstanden hat (deswegen ist der [415 Unsupported Media Type Status Code](https://dataloft.ch/wiki/415-http-statuscodes-unsupported-media-type/) unangemessen) und die Syntax der Anfrage-Entität ist korrekt (weswegen auch kein [400 Bad Request Status Code](https://dataloft.ch/wiki/400-http-statuscodes-bad-request/) angemessen ist), war aber dennoch nicht in der Lage die innewohnenden Instruktionen zu verarbeiten. Dieser Fehlerzustand kann zum Beispiel auftreten, wenn ein XML-Anfrage-Körper wohlgeformte (dh. syntaktisch korrekte), aber semantisch fehlerhafte XML-Instruktionen enthält."

Reply:
I will check if i found something that match better.

Reply:
409 conflict sounds good, what do you mean?